### ISSUE TITLE
[Enhancement]: Product Collection > Image > display out of stock badge

### ISSUE THREAD

**Issue by Marc-pi** [woothemes/woocommerce#42620]
### Describe the solution you'd like

In FSE context (yes, some are making website with full FSE ;-) ), if we make use of the WP standard Query Loop Block, 
filtered on Product type, "out of stock badges" are not displayed in results.

This badge is very usefull in an eCommerce context, for marketing purposes

Proposed solution : extend the WP Query Loop to add a toggle in settings "Display Out of stock badge"
to make Woo the best showcase for early FSE adopters like use

### Describe alternatives you've considered

see https://github.com/woocommerce/woocommerce/issues/42621 : i tried to make use of Newest Products block, but the badge is not displayed either

**Comment by rrennick** [comment]
@Marc-pi Thanks for creating the issue. I'll transfer this issue to the block repo where blocks/FSE support is maintained.

**Comment by gigitux** [comment]
Thank you for bringing up this issue!

I would like to understand the reasoning behind choosing the classic Query Loop block instead of the Products block. Could you please provide more context or specific use cases?

Additionally, it would be helpful if you could clarify the desired outcome for the implementation. In case you wish to display the stock indicator, you can achieve this by using the `Product Stock Indicator` block:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/5b2c6a7e-bc82-4721-93b6-096d6097d391)

**Comment by imanish003** [comment]
Hi @Marc-pi,

As suggested by @gigitux, incorporating the `Product Stock Indicator` block is a fantastic way to display the stock status. To assist you further, I've prepared a brief demo for your reference:

https://github.com/woocommerce/woocommerce-blocks/assets/16707866/075305c9-022f-497a-9c08-0181276f83bb

Since we haven't received any follow-up from your end in the past two weeks, we'll be closing this issue for now. However, please bear in mind that we're always open to revisiting and reopening the discussion once we receive more insights from your side.

Thank you for your involvement, and please don't hesitate to share more thoughts whenever you're ready.

**Comment by Marc-pi** [comment]
> I would like to understand the reasoning behind choosing the classic Query Loop block instead of the Products block. 

because at that time (2022, July) the woo loopers was not available or ready. 
I tested other alternatives like the Newest Products block (see above and 
and  https://github.com/woocommerce/woocommerce/issues/42621

> As suggested by @gigitux, incorporating the Product Stock Indicator block is a fantastic way to display the stock status

ok, thanks for the tip, i did not notice this block was there or newly shipped (WooCommerce Blocks 10.1.0 released April 2023) !!!
Since i'm talking here about a badge (see issue name), i would imagine something displayed "on" the image, like frequently on other templates (it is more natural and keep the UX light). 

**So i thought more about a block setting option**

to be displayed like this one

![image](https://github.com/woocommerce/woocommerce-blocks/assets/919332/c9e41ae8-0465-416d-8402-31e1d0f71b5b)

**Comment by Marc-pi** [comment]
i did not notice such an option on the Products (beta) block (display the "Out of Stock" badge on the image),
'm not able to test for the new loop block Product Collection
please reopen the issue @imanish003 if it is not managed by the new one

**Comment by imanish003** [comment]
Hello @Marc-pi,

I wanted to let you know that I've reopened the issue. The functionality you mentioned isn't actually supported by either the Product Collection or Products beta block. Thank you for bringing this up.

**Comment by Marc-pi** [comment]
for the setting on the Image Block, see https://github.com/woocommerce/woocommerce/issues/42621

**Comment by Marc-pi** [comment]
there is a On-Sale Badge Block https://github.com/woocommerce/woocommerce/issues/55996
Inner Block in Product Image Block is now supported https://github.com/woocommerce/woocommerce/pull/57586

but we miss a solid Sold Out Block
i've tested with the product stock indicator, but it is not displayed (Woo 9.8x)

**Comment by gigitux** [comment]
> there is a On-Sale Badge Block [#55996](https://github.com/woocommerce/woocommerce/issues/55996) Inner Block in Product Image Block is now supported [#57586](https://github.com/woocommerce/woocommerce/pull/57586)
> 
> but we miss a solid Sold Out Block i've tested with the product stock indicator, but it is not displayed (Woo 9.8x)

Are you sure? Can you provide some reproduction steps? I'm able to see it:

| Editor | Frontend |
|--------|--------|
| ![Image](https://github.com/user-attachments/assets/f3363282-a6f8-4432-a275-6f6b7be2cd32)| ![Image](https://github.com/user-attachments/assets/2d0dd804-ad48-48fe-81ec-79dbd78ab5cd) | 


The `product stock indicator` is is hidden when products are in stock.

**Comment by Marc-pi** [comment]
> The product stock indicator is is hidden when products are in stock. 

not exactly. It displays the stock quantity when there is stock (when stock qty is managed), as per its name (stock indicator, not stock status)
the need is to be able to only display the Out of Stock status, to have a badge.
A display above the product image would be fine, aside with the Sale badge.
A block option would be fine


![Image](https://github.com/user-attachments/assets/c86115a0-b490-4981-86eb-2d8cef8ec7cd)

when no price is set

![Image](https://github.com/user-attachments/assets/df859b1d-29c8-4016-a2db-7b3f44302c72)

**Comment by kkoteen** [comment]
Hi @gigitux could you take another look at this issue and recommend next steps? Thanks!

**Comment by gigitux** [comment]
I'm going to close this issue as fixed because it is possible to add any block on top of the Product Image block. So, it is possible to add the Product Stock Indicator on top of the Product Image block:

https://github.com/user-attachments/assets/5ef93347-0a7d-4f4a-8fb8-a8ad93b91e21
